packages/x-charts/src/ChartsLegend/DefaultChartsLegend.tsx

```diff
@@ -3,14 +3,28 @@ import PropTypes from 'prop-types';
 import { FormattedSeries } from '../context/SeriesProvider';
 import { LegendPerItem, LegendPerItemProps } from './LegendPerItem';
 import { DrawingArea } from '../context/DrawingProvider';
+import { SeriesLegendItemContext } from './chartsLegend.types';
 
-export interface LegendRendererProps extends Omit<LegendPerItemProps, 'itemsToDisplay'> {
+export interface LegendRendererProps
+  extends Omit<LegendPerItemProps, 'itemsToDisplay' | 'onItemClick'> {
   series: FormattedSeries;
   seriesToDisplay: LegendPerItemProps['itemsToDisplay'];
   /**
    * @deprecated Use the `useDrawingArea` hook instead.
    */
   drawingArea: Omit<DrawingArea, 'isPointInside'>;
+  /**
+   * Callback fired when a legend item is clicked.
+   * @param {React.MouseEvent<SVGRectElement, MouseEvent>} event The click event.
+   * @param {SeriesLegendItemContext} legendItem The legend item data.
+   * @param {number} index The index of the clicked legend item.
+   * @default undefined
+   */
+  onItemClick?: (
+    event: React.MouseEvent<SVGRectElement, MouseEvent>,
+    legendItem: SeriesLegendItemContext,
+    index: number,
+  ) => void;
 }
 
 function DefaultChartsLegend(props: LegendRendererProps) {
```

Comment:
Does this component provide any value? 🤔 

It feels like we could just remove it to simplify the legend setup a bit in V8

Reply:
It does not. This `ChartsLegend`/`DefaultChartsLegend` are here such that you can either:

- Use `<ChartsLegend/>` in a composed component
- Import `DefaultChartsLegend`, style it, and pass it to `slots.legend`
- Create a custom legend component pass it to `slots.legend` (It's the `ChartsLegend` that take care of passing the props)

It's like the tooltip with a main component doing the processing, and a second on doing the rendering. The second one being the one you can modify with the `slots`.

Honestly, this feels bad, and a solution with `useLegend()` to allow people to creat their own Legend will feel much better

Reply:
It took me quite a while to understand the order of components 😢

Reply:
https://github.com/mui/mui-x/issues/13819 leave this ref here so we don't forget 😅